### PR TITLE
Read more LDAP attributes to fill the user model

### PR DESCRIPTION
**Pull request by DeepDiver1975**
Screenshot from 2019-12-06 11-08-07

### CODE REVIEW

File: pkg/service/v0/service.go [hunk 1]
```diff
@@ -30,8 +31,15 @@ func NewService(opts ...Option) Service {
...
 	}
 
-	m.HandleFunc("/v1.0/me", svc.Me)
-	m.HandleFunc("/v1.0/users", svc.Users)
+	m.Route("/v1.0", func(r chi.Router) {
+		r.Get("/me", svc.Me)
+		r.Route("/users", func(r chi.Router) {
+			r.Get("/", svc.Users)
+			r.Route("/{userId}", func(r chi.Router) {
+				r.Get("/", svc.Users)
```
Comment (tboerger) [inline pkg/service/v0/service.go]: It's more common to have a separate handler for that, just add the definition to the interface and all service implementations.

Comment (DeepDiver1975) [reply pkg/service/v0/service.go]: Sure - I'll add some more refactorings .... need to extract all ldap stuff out of service.go

File: pkg/service/v0/graph.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+// UserCtx middleware is used to load an User object from
+// the URL parameters passed through as the request. In case
+// the User could not be found, we stop here and return a 404.
+func (g Graph) UserCtx(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var user *ldap.Entry
+		var err error
+
+		if userID := chi.URLParam(r, "userID"); userID != "" {
+			user, err = g.ldapGetUser(userID)
```
Comment (refs) [inline pkg/service/v0/graph.go:39]: `err` is not handled

File: pkg/service/v0/graph.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+			"entryuuid",
+			"sn",
+		},
+		nil,
+	)
+
+	return con.Search(search)
+}
+
+func createUserModel(displayName string, id string) *msgraph.User {
```
Comment (refs) [inline pkg/service/v0/graph.go:168]: personally I'd move this and `createUserModelFromLDAP` method to an utils file (same package)

**Review by refs — APPROVED** [review]
👍 LGTM. ~I would consider using [ldappool](https://github.com/vetinari/go-ldappool) adapted to `v3`, if possible, if you want to reuse connections~ https://github.com/go-ldap/ldap/pull/48

**Pull request merged**